Stratis.Bitcoin/Features/BlockStore/LoopSteps/BlockStoreLoopStep.cs

```diff
@@ -0,0 +1,87 @@
+﻿using NBitcoin;
+using Stratis.Bitcoin.Utilities;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Stratis.Bitcoin.Features.BlockStore.LoopSteps
+{
+    internal sealed class BlockStoreLoopStepChain
+    {
+        private List<BlockStoreLoopStep> steps = new List<BlockStoreLoopStep>();
+        private bool disposeMode;
+
+        public BlockStoreLoopStepChain(bool disposeMode)
+        {
+            this.disposeMode = disposeMode;
+        }
+
+        internal void SetNextStep(BlockStoreLoopStep step)
+        {
+            this.steps.Add(step);
+        }
+
+        internal async Task<BlockStoreLoopStepResult> Execute(ChainedBlock nextChainedBlock, CancellationToken cancellationToken)
+        {
+            var result = new BlockStoreLoopStepResult().Next();
+
+            foreach (var step in this.steps)
+            {
+                var stepResult = await step.Execute(nextChainedBlock, cancellationToken, this.disposeMode);
+
+                if (stepResult.ShouldBreak)
+                {
+                    result = stepResult;
+                    break;
+                }
+
+                if (stepResult.ShouldContinue)
+                {
+                    result = stepResult;
+                    break;
```

Comment:
The break is correct... cause the DownloadAndStoreBlocks() method will decide what to do with the result. I have refactored this little bit of code to be:

``if (stepResult.ShouldBreak || stepResult.ShouldContinue) 
{
                    result = stepResult;
                    `break;